Source/Shaders/Builtin/Functions/sampleOctahedralProjection.glsl

```diff
@@ -1,42 +1,74 @@
 /**
- * Samples from a cube map that has been projected using an octahedral projection from the given direction.
- *
- * @name czm_sampleOctahedralProjection
- * @glslFunction
- *
- * @param {sampler2D} projectedMap The texture with the octahedral projected cube map.
- * @param {vec2} textureSize The width and height dimensions in pixels of the projected map.
- * @param {vec3} direction The normalized direction used to sample the cube map.
- * @param {float} lod The level of detail to sample.
- * @returns {vec3} The color of the cube map at the direction.
+ * Samples the 4 neighboring pixels and return the weighted average. 
+ * 
+ * @private
  */
-vec3 czm_sampleOctahedralProjection(sampler2D projectedMap, vec2 textureSize, vec3 direction, float lod) {
+vec3 czm_sampleOctahedralProjectionWithFiltering(sampler2D projectedMap, vec2 textureSize, vec3 direction, float lod)
+{
     direction /= dot(vec3(1.0), abs(direction));
     vec2 rev = abs(direction.zx) - vec2(1.0);
     vec2 neg = vec2(direction.x < 0.0 ? rev.x : -rev.x,
                     direction.z < 0.0 ? rev.y : -rev.y);
     vec2 uv = direction.y < 0.0 ? neg : direction.xz;
     vec2 coord = 0.5 * uv + vec2(0.5);
-
-    lod = floor(lod);
+    vec2 pixel = 1.0 / textureSize;
 
     if (lod > 0.0)
     {
         // Each subseqeuent mip level is half the size
         float scale = 1.0 / pow(2.0, lod);
         float offset = ((textureSize.y + 1.0) / textureSize.x);
-        vec2 pixel = 1.0 / textureSize;
 
         coord.x *= offset;
         coord *= scale;
 
         coord.x += offset + pixel.x;
         coord.y += (1.0 - (1.0 / pow(2.0, lod - 1.0))) + pixel.y * (lod - 1.0) * 2.0;
+
+        pixel *= scale;
     }
     else
     {
         coord.x *= (textureSize.y / textureSize.x);
     }
 
-    return texture2D(projectedMap, coord).rgb;
+    // Do bilinear filtering
+    vec3 color1 = texture2D(projectedMap, coord + vec2(0.0, pixel.y)).rgb;
```

Comment:
This should only happen if the `OES_texture_float_linear` extension isn't available. Do you have that extension available?

Reply:
I do have that extension available. Should it just return `return texture2D(projectedMap, coord).rgb;` in that case?